content/scout/integrations/ci/jenkins.md

```diff
@@ -5,22 +5,34 @@ title: Integrate Docker Scout with Jenkins
 ---
 
 You can add the following stage and steps definition to a `Jenkinsfile` to run
-Docker Scout as part of a Jenkins pipeline. The pipeline needs two secrets
-defined to authenticate with Docker Hub: `DOCKER_HUB_USER` and `DOCKER_HUB_PAT`
-It also needs an environment variable defined for the image and tag.
+Docker Scout as part of a Jenkins pipeline. The pipeline needs a `DOCKER_HUB`
+credential containing the username and password for authenticating to Docker
+Hub. It also needs an environment variable defined for the image and tag.
 
 ```groovy
-…
-stage('Analyze image') {
-    steps {
-        // Install Docker Scout
-        sh 'curl -sSfL https://raw.githubusercontent.com/docker/scout-cli/main/install.sh | sh -s -- -b /usr/local/bin'
-        
-        // Log into Docker Hub
-        sh 'echo $DOCKER_HUB_PAT | docker login -u $DOCKER_HUB_USER --password-stdin'
+pipeline {
+    agent {
+        // Agent details
+    }
+
+    environment {
+        DOCKER_HUB = credentials('jenkins-docker-hub-credentials')
```

Comment:
**question:** what does this assignment actually do? 🤔 Is there something internally that is making use of `$DOCKER_HUB`? Or doesn't it really matter and all we want is for `$DOCKER_HUB_PSW` and `$DOCKER_HUB_USR` to be set by this function? Working through side-effects is a bit odd (for a Jenkins layman like myself)

Reply:
The credentials function creates the USR and PSW suffix variables

Reply:
So what does `DOCKER_HUB` contain? Why do we assign it to a variable? 🤔 Is that something we expect users to know?

Reply:
yeah I would expect a Jenkins user to be familiar with the credentials method https://www.jenkins.io/doc/book/pipeline/jenkinsfile/#usernames-and-passwords